### PR TITLE
feat(bridge): allow user-defined `scriptSetup` options

### LINKED ISSUE #2184: Nuxt `scriptSetup` options are ignored
### Environment

- Operating System: `Darwin`
- Node Version:     `v16.11.1`
- Nuxt Version:     `2.16.0-27299536.ab1c6cb4`
- Package Manager:  `yarn@1.22.17`
- Bundler:          `Webpack`
- User Config:      `bridge`, `components`, `autoImports`
- Runtime Modules:  `-`
- Build Modules:    `@vueuse/core/nuxt@7.1.2`, `@nuxt/bridge@3.0.0-27296423.f3082ca`

### Reproduction

Create a Nuxt app with Nuxt Bridge and set the `scriptSetup` option in `nuxt.config.ts` as defined on the [unplugin-vue2-script-setup](https://github.com/antfu/unplugin-vue2-script-setup) repo.

### Describe the bug

The options are discarded because they are not passed to the vite/webpack transform plugin context.

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by yassilah**
### 🔗 Linked issue

#2184

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves  #2184 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/bridge/src/setup.ts [hunk 1]
```diff
@@ -3,9 +3,10 @@ import scriptSetupPlugin from 'unplugin-vue2-script-setup'
 
 export const setupScriptSetup = () => {
   const nuxt = useNuxt()
-
-  addVitePlugin(scriptSetupPlugin.vite())
-  addWebpackPlugin(scriptSetupPlugin.webpack())
+  const options = nuxt.options.scriptSetup
```
Comment (pi0) [inline packages/bridge/src/setup.ts:6]: We should read and pass option from `bridge.scriptSetup` [here](https://github.com/nuxt/framework/blob/f9ed010faa9e22af4b87e9bea24ad6a929bf7e6d/packages/bridge/src/module.ts#L48)

Comment (yassilah) [reply packages/bridge/src/setup.ts]: You're right, I also added the types from `unplugin-vue2-script-setup` to set the options in the bridge config.

File: packages/bridge/types.d.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
 import type {} from '@nuxt/nitro'
 import type { NuxtConfig as _NuxtConfig } from '@nuxt/schema'
 import type { MetaInfo } from 'vue-meta'
+import { PluginOptions as ScriptSetupPluginOptions } from 'unplugin-vue2-script-setup/dist'
```
Comment (pi0) [inline packages/bridge/types.d.ts:4]: @antfu Any chance we can have `PluginOptions` types from default export?

Comment (pi0) [inline packages/bridge/types.d.ts:4]:
```suggestion
import type { PluginOptions as ScriptSetupPluginOptions } from 'unplugin-vue2-script-setup/dist'
```

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks!

**Pull request merged**